pages/api/book/[user].ts

```diff
@@ -38,19 +59,38 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     }
   });
 
-  const evt: CalendarEvent = {
+  let rawLocation = req.body.location
+
+  let evt: CalendarEvent = {
     type: selectedEventType.title,
     title: getEventName(req.body.name, selectedEventType.title, selectedEventType.eventName),
     description: req.body.notes,
     startTime: req.body.start,
     endTime: req.body.end,
-    location: req.body.location,
     organizer: {email: currentUser.email, name: currentUser.name, timeZone: currentUser.timeZone},
     attendees: [
       {email: req.body.email, name: req.body.name, timeZone: req.body.timeZone}
     ]
   };
 
+  // If phone or inPerson use raw location
+  // set evt.location to req.body.location
+  if (!rawLocation.includes('integration')) {
+    evt.location = rawLocation
+  }
+  
+
+  // If location is set to an integration location
+  // Build proper transforms for evt object
+  // Extend evt object with those transformations
+  if (rawLocation.includes('integration')) {
+    let maybeLocationRequestObject = getLocationRequestFromIntegration({
+      location: rawLocation
+    }) 
+    
+    evt = merge(evt, maybeLocationRequestObject)
```

Comment:
Instead of using Lodash you can use { ...evt, ...maybeLocationRequestObject } here

Reply:
If I remember correctly, spreading on objects only does a shallow clone.